### ISSUE TITLE
[resources/templates] Ability to ignore some files

### ISSUE THREAD

**Issue by mathieumd** [debops/debops#2345]
I'm using the `resources`' `templates/by-{group,host}/` path a lot to easily track and deploy files to servers.

However, it also send [Vim's swap files](https://www.baeldung.com/linux/vim-swap-files), which sometimes break Jinja2 templating engine (see below).

Is there a way to ignore some files to be sync'ed through `resources`' `templates`?

Example of breaking Jinja2 templating:
```
"src": "/home/mathieu/dev/ansible/resources/templates/by-group/squid_servers/var/www/html/.config.pac.swp",
[...],
"msg": "AnsibleError: template error while templating string: unexpected char \'!\' at 12119.
String: b0VIM 8.2\\u0000\\u0000\\u0000\\u0000\\u0010\\u0000[...]
```

Of course, in the meantime, I simply changed Vim's temporary saving path (see [Change Default Storage Location](https://www.baeldung.com/linux/vim-swap-files#change-default-storage-location)). But still I think adding ignoring feature to `resources` could be helpful.

**Comment by drybjed** [comment]
There might be a way via conditionally checking if a given item's filename ends with a string from a list of ignored extensions, but I'm not sure if this will be triggered before the template error during the `community.general.filetree` lookup.

**Comment by mathieumd** [comment]
Yes, looks like it would work: I tested by adding `and not item.src|regex_search('/\.[^/]+\.sw[l-p]$')` at [resources/tasks/main.yml#L39](https://github.com/debops/debops/blob/master/ansible/roles/resources/tasks/main.yml#L39) and it nicely ignore my `.temp-test.swp` containing a broken `{{test`.
